celavi/routing.py

```diff
@@ -193,199 +191,125 @@ def get_all_routes(
-            ]
-
-            # Remove entries from the out-of-state route list where the
-            # connections are required to be in-state
-            _keep = outstate_routes.in_state_only == False
-            #route_list = instate_routes.append(outstate_routes[_keep])
-            route_list = pd.concat([instate_routes,outstate_routes[_keep]])
-
-            # if route_list is empty, generate empty data frame for export (e.g., create column for total_vkmt)
-            # otherwise, loop through all locations in route_list and compute routing distances
-            if route_list.empty:
-                print("list is empty")
-                route_list["vkmt"] = ""
-                route_list = route_list.drop(["merge"], axis=1)
-                route_list["total_vkmt"] = route_list.groupby(
-                    by=[
-                        "source_facility_id",
-                        "source_facility_type",
-                        "source_lat",
-                        "source_long",
-                        "destination_facility_id",
-                        "destination_facility_type",
-                        "destination_lat",
-                        "destination_long",
-                    ]
-                )["vkmt"].transform("sum")
-                route_list.to_csv(file_output)
-
-            else:
-
-                route_list.to_csv(routing_output_folder + "route_list.csv")
-
-                _routes = route_list[
-                    ["source_long", "source_lat", "destination_long", "destination_lat"]
-                ]
-
-                _routes.to_csv(routing_output_folder + "latlongs.csv")
-
-                # if routing engine is specified, use it to get the route (fips and
-                # vkmt) for each pair of input locations
-                if router is not None:
-
-                    # initialize holder for all routes
-                    _vkmt_by_county_all_routes = pd.DataFrame()
-
-                    # loop through all locations to compute routes
-                    print("finding routes", flush=True)
-                    for i in np.arange(_routes.shape[0]):
-
-                        # print every 20 routes
-                        if i % 20 == 0:
-                            print(i, flush=True)
-
-                        _vkmt_by_county = router.get_route(
-                            start=(
-                                _routes.source_long.iloc[i],
-                                _routes.source_lat.iloc[i],
-                            ),
-                            end=(
-                                _routes.destination_long.iloc[i],
-                                _routes.destination_lat.iloc[i],
-                            ),
+            # Save the raw routes file including edges that violate the vkmt_max
+            network_all_routes.to_csv('network-routes-all.csv', index=False)
+
+            # Remove edges from the network if any have distances greater than the max allowed
```

Comment:
Same as comment above. Would like to learn more. For instance, how does this differentiate between routes that are totally useless and a unique route that is above vkmt_max but is the only option?

...just read 5 lines below Julien... -_-